src/Numberable.ts

```diff
@@ -1,68 +1,77 @@
 class Numberable extends Number {
-  private _isPercent: boolean
+  ["constructor"]!: typeof Numberable;
+
+  private _isPercent: boolean;
   constructor(number, options: any = {}) {
-    super(number)
+    super(number);
 
-    const { isPercent = false } = options
-    this._isPercent = isPercent
+    const { isPercent = false } = options;
+    this._isPercent = isPercent;
   }
 
   of(number) {
     if (this._isPercent) {
-      return new (this as any).constructor(this as any * number)
+      return new this.constructor((this as any) * number);
     }
-    return new (this as any).constructor(this as any / number)
+    return new this.constructor((this as any) / number);
   }
 
   percent() {
-    return new (this as any).constructor(this as any / 100, { isPercent: true })
+    return new this.constructor((this as any) / 100, { isPercent: true });
   }
 
   inPercent() {
-    return new (this as any).constructor(this as any * 100)
+    return new this.constructor((this as any) * 100);
   }
 
   ordinal() {
-    const finalDigit = this.toString().slice(-1)
-    return this + (['th', 'st', 'nd', 'rd'][finalDigit] || 'th')
+    const finalDigit = this.toString().slice(-1);
+    return ((this +
+      (["th", "st", "nd", "rd"][finalDigit] || "th")) as unknown) as Numberable;
```

Comment:
return type is actually a string.

TS can type this by itself if we properly convert the number to a string first:
`return this.toString() +  (["th", "st", "nd", "rd"][finalDigit] || "th")`